config/crds/apis.kcp.dev_apibindings.yaml-patch

```diff
@@ -6,3 +6,9 @@
   path: /spec/versions/name=v1alpha1/schema/openAPIV3Schema/properties/spec/properties/reference/properties/workspace/oneOf
   value:
   - required: ["path"]
+- op: add
+  path: /spec/versions/name=v1alpha1/schema/openAPIV3Schema/properties/spec/properties/acceptedPermissionClaims/items/x-kubernetes-validations
+  value:
+    - rule: |
+        (((has(self.group) && self.group == '') || !(has(self.group))) && self.resource in ['configmaps', 'namespaces', 'secrets', 'serviceaccounts']) || ((has(self.group) && self.group != '') && (has(self.identityHash) && self.identityHash != ''))
```

Comment:
why is that a giant expression? can be split up, giving better messages, and would be much easier to read.

Reply:
I do not have a way to do it because the rules are ANDed together AFAICT

Reply:
I don't see why ANDing should stop that from working. Just turn it into a disjunctive normal form.
```
- rule: "(has(self.group) && self.group != '') || self.resource in ['configmaps', 'namespaces', 'secrets', 'serviceaccounts']"
  message: .resource must be one of configmaps, namespaces, secrets, serviceaccounts if .group is empty.
- rule: "!(has(self.group) && self.group != '') || (has(self.identityHash) && self.identityHash != '')
  message: .identityHash must be set if .group is not empty.
```